// film-node/lib/client_response.js
var config = require('./config.json');

// tranquansp
// 9 Aug 2014
// generate 404 Page (because of wrong request)

var html_Page404 = function()
{
	var Page404 = '<div style="margin-top:50px;text-align:center;"><h1>Page not found<h1><h2><a href="/">Quay về Trang Chủ</a></h2></div>';
	return Page404;
}

// tranquansp
// 9 Aug 2014
// put data (array of film from collection) to content and generate html to client

var html_addData_To_IndexPage = function(response,items)
{
	if(items)
	{
		// put data to template :
		// + items
		// + css File adress
		// + link to point to pagecurrent (view detail 1 film)
		// + link to point to image directory
		// at this moment : phim in index Page that is current Films, dont worry , we could change later
		response.render(config.EJS_INDEX_PAGE,{ title :" ::. Review Phim Chiếp Rạp .::. Rủ rê nhau xem phim .:: ", cssFile : config.FILE_STYLE_CSS , 
										content : items , prelink : config.PAGE_CURR + config.SYM_SEPARATE ,
										fullprelink : config.HTTP_STRING + config.WEBSERVER_HOST + config.SYM_COLON + config.WEBSERVER_PORT + config.PAGE_CURR + config.SYM_SEPARATE,
										imgDir : config.PUBLIC_DIR + config.IMAGE_DATA_DIR + config.SYM_SEPARATE });
	}
};

var html_addGeneralDataToContent = function(response,items)
{
	if(items)
	{
		// put data to template :
		// + items
		// + css File adress
		// + link to point to pagecurrent (view detail 1 film)
		// + link to point to image directory
		response.render(config.EJS_CURR_FILMS_PAGE,{ title :" ::. Các phim đang chiếu rạp .:: ", cssFile : config.FILE_STYLE_CSS , 
										content : items , prelink : config.PAGE_CURR + config.SYM_SEPARATE ,
										fullprelink : config.HTTP_STRING + config.WEBSERVER_HOST + config.SYM_COLON + config.WEBSERVER_PORT + config.PAGE_CURR + config.SYM_SEPARATE,
										imgDir : config.PUBLIC_DIR + config.IMAGE_DATA_DIR + config.SYM_SEPARATE });
	}
};



// tranquansp
// 9 Aug 2014
// put detail data to content of current films and generate html to client

var html_addDetailData_To_ContentCurrentFilm = function(response,item)
{
	if(item)
	{
		// put detail data to template :
		// + items
		// + css File adress
		// + fullprelink to point to page current films
		// + imgDir to point to image directory
		response.render(config.EJS_DETAIL_PAGE,{ title :item.name, cssFile : config.FILE_STYLE_CSS , 
										item : item , fullprelink : config.HTTP_STRING + config.WEBSERVER_HOST + config.SYM_COLON + config.WEBSERVER_PORT + config.PAGE_CURR + config.SYM_SEPARATE ,
										imgDir : config.PUBLIC_DIR + config.IMAGE_DATA_DIR + config.SYM_SEPARATE });										
	}
};

// tranquansp
// 9 Aug 2014
// client chats with other clients

var html_showChatClientPage = function(response)
{
		response.render(config.EJS_CHAT_PAGE,{ title : "Rủ rê đi xem phiêm " });										
};

//##### Export for using in real App	
exports.html_addGeneralDataToContent = html_addGeneralDataToContent;
exports.html_addDetailData_To_ContentCurrentFilm = html_addDetailData_To_ContentCurrentFilm;
exports.html_Page404 = html_Page404;
exports.html_showChatClientPage = html_showChatClientPage;
exports.html_addData_To_IndexPage = html_addData_To_IndexPage;
