// film-node/lib/film-node.js
/**************************************************/
//                      INIT                      //
/*************************************************/
var config = require('./config.json');
var clientRes = require('./client_response'),
	socketWorker = require('./socket');
    fs = require('fs');
var requestWorker = require('cluster');
var MongoClient = require('mongodb').MongoClient,
    Server = require('mongodb').Server,
    db;

// mongo config
var mongoOptions = {
  'auto_reconnect': true,
  'poolSize': config.MONGO_MAX_POOLSIZE
};

// init collection, reuse later
var collection_cl_currentFilms,collection_cl_generalFilms,collection_cl_feedback;

var mongoClient = new MongoClient(new Server(config.MONGO_HOST, config.MONGO_PORT,mongoOptions));
mongoClient.open(function(err, mongoClient) {
    db = mongoClient.db(config.MONGO_DB_NAME);
    collection_cl_currentFilms = db.collection(config.MONGO_CL_CURRENTFILMS);
    collection_cl_generalFilms = db.collection(config.MONGO_CL_ALLFILMS);
    collection_cl_feedback = db.collection(config.MONGO_CL_FEEDBACK);
});

/*************************************************/
//.........Show data on index Page...............//
/*************************************************/
// tranquansp
// 9 Aug 2014
// show data on index Page = current films 

var html_showDataIndexPage = function(request, response) {

    //db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {
		collection_cl_currentFilms.find().toArray(function(err, items) {
			if(err) throw err;
			console.log('[index Page] Worker express core ' + requestWorker.worker.id + ' is working');
			clientRes.html_addData_To_IndexPage(response,items);
		});
    //});
};
/*************************************************/
//.........Return ajax data CurrFilm Page...........//
/*************************************************/
// tranquansp
// 13 Aug 2014
// show data on index Page = current films 

var html_showAjaxCurrFilmPage = function(request, response) {
	//~ console.log('come here');
    //db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {
		//~ collection_cl_currentFilms.find().toArray(function(err, items) {
			//~ if(err) throw err;
			//~ console.log('[index Page] Worker express core ' + requestWorker.worker.id + ' is working');
			//~ //clientRes.html_addData_To_IndexPage(response,items);
			//~ response.send(items);
		//~ });
    //});
};

/*************************************************/
//...Show data on Phim dang chieu Page............/
/*************************************************/
// tranquansp
// 9 Aug 2014
// show data on current films Page

var html_showDataByClCurrFilm = function(request, response) {
	//db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {
		collection_cl_currentFilms.find().toArray(function(err, items) {
			if(err) throw err;
			clientRes.html_addGeneralDataToContent(response,items);
		});
	//});	
};


/*************************************************/
//.........Show detail of Film..................../
/*************************************************/
// tranquansp
// 9 Aug 2014
// show detail of 1 films

var html_showDataCurrentFilmById = function(request, response) {
	try{
		// hard code : request.params.param_1, turn back to file app.js to understand
		var value = parseInt(request.params.param_1);
		//db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {

				if(!isNaN(value)){
					var query = {};
					query[config.FILM_ID] = value;
					collection_cl_currentFilms.findOne(query,function(err,item){
						if (err) throw err;
						if(item != null){
							clientRes.html_addDetailData_To_ContentCurrentFilm(response,item);
						}
						else
							response.send(clientRes.html_Page404());
					});
				}
				else
					response.send(clientRes.html_Page404());
		//});
	}
	catch(e)
	{
		console.log(e);
		response.send(clientRes.html_Page404());
	}
};	

/*************************************************/
//.................Chat module..................../
/*************************************************/
// tranquansp
// 9 Aug 2014
// Chat module with socket,express,javascript

var html_chatPage = function(request, response) {
	
	clientRes.html_showChatClientPage(response);
};


/*************************************************/
//.........Show content of contact Page.........../
/*************************************************/
var html_showContentContactPage = function(request, response) {	
		response.render(config.EJS_CONTACT_PAGE,{ title :" Liên hệ với chúng tôi "});
};

/*************************************************/
//................Other codes....................//
/*************************************************/

var html_showDataNewFilmById = function(request, response) {
};

var html_showDataOldFilmById = function(request, response) {
};

var html_showDataAllFilmById = function(request, response) {
	response.send(clientRes.html_Page404());
};

var html_tempPage = function(request, response) {
	response.send(clientRes.html_Page404());
};

// chat Page task
var socket_Tasks = function(io){
	//console.log('[index Page] Worker socket core : ' + requestWorker.worker.id + ' is working');
	socketWorker.socket_communication(io,requestWorker);
}

/*************************************************/
//................EXPORT FUNC....................//
/*************************************************/
exports.html_showDataCurrentFilmById = html_showDataCurrentFilmById;
exports.html_showDataOldFilmById = html_showDataOldFilmById;
exports.html_showDataNewFilmById = html_showDataNewFilmById;

exports.html_showDataAllFilmById = html_showDataAllFilmById;
exports.html_showDataByClCurrFilm = html_showDataByClCurrFilm;

exports.html_tempPage = html_tempPage;
exports.html_showDataIndexPage = html_showDataIndexPage;
exports.html_showContentContactPage = html_showContentContactPage;
exports.html_chatPage = html_chatPage;
exports.socket_Tasks = socket_Tasks;

exports.html_showAjaxCurrFilmPage = html_showAjaxCurrFilmPage;
