// film-node/lib/film-node.js
/**************************************************/
//                      INIT                      //
/*************************************************/
var config = require('./config.json');
var clientRes = require('./client_response');

var MongoClient = require('mongodb').MongoClient,
    Server = require('mongodb').Server,
    db;

var mongoClient = new MongoClient(new Server(config.MONGO_HOST, config.MONGO_PORT));
mongoClient.open(function(err, mongoClient) {
    db = mongoClient.db(config.MONGO_DB_NAME);
});


/*************************************************/
//.........Show data on index Page...............//
/*************************************************/

var html_showDataIndexPage = function(request, response) {
    db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {
		collection.find().toArray(function(err, items) {
			if(err) throw err;
			var html_res = clientRes.html_addDataToHeader()
				+ clientRes.html_addGeneralDataToContent(items)
				+ clientRes.html_addDataToFooter();
			response.send(html_res);
		});
    });
};

/*************************************************/
//...Show data on Phim dang chieu Page............/
/*************************************************/

var html_showDataByClCurrFilm = function(request, response) {
	db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {
		collection.find().toArray(function(err, items) {
			if(err) throw err;
			var html_res = clientRes.html_addDataToHeader()
				+ clientRes.html_addGeneralDataToContent(items)
				+ clientRes.html_addDataToFooter();
			response.send(html_res);
		});
	});	
};

/*************************************************/
//.........Show detail of Film..................../
/*************************************************/


var html_showDataCurrentFilmById = function(request, response) {
	try{
		// hard code : request.params.param_1, turn back to file app.js to understand
		var value = parseInt(request.params.param_1);
		db.collection(config.MONGO_CL_CURRENTFILMS, function(err, collection) {

				if(!isNaN(value)){
					var query = {};
					query[config.FILM_ID] = value;
					collection.findOne(query,function(err,item){
						if (err) throw err;
						if(item != null){
							var html_res = clientRes.html_addDataToHeader()
							+ clientRes.html_addDetailDataToContent(item)
							+ clientRes.html_addDataToFooter();
							response.send(html_res);
						}
						else
							response.send(clientRes.html_Page404());
					});
				}
				else
					response.send(clientRes.html_Page404());
		});
	}
	catch(e)
	{
		console.log(e);
		response.send(clientRes.html_Page404());
	}
};	

/*************************************************/
//................Other codes....................//
/*************************************************/

var html_showDataNewFilmById = function(request, response) {
};

var html_showDataOldFilmById = function(request, response) {
};

var html_showDataAllFilmById = function(request, response) {
	response.send(clientRes.html_Page404());
};

var html_tempPage = function(request, response) {
	response.send(clientRes.html_Page404());
};

/*************************************************/
//................EXPORT FUNC....................//
/*************************************************/
exports.html_showDataCurrentFilmById = html_showDataCurrentFilmById;
exports.html_showDataOldFilmById = html_showDataOldFilmById;
exports.html_showDataNewFilmById = html_showDataNewFilmById;

exports.html_showDataAllFilmById = html_showDataAllFilmById;
exports.html_showDataByClCurrFilm = html_showDataByClCurrFilm;

exports.html_tempPage = html_tempPage;
exports.html_showDataIndexPage = html_showDataIndexPage;
